tools/launch.ts

```diff
@@ -0,0 +1,362 @@
+        choices: [
+          "rococo",
+          "westend",
+          "kusama",
+          "polkadot",
+          "rococo-local",
+          "westend-local",
+          "kusama-local",
+          "polkadot-local",
+        ],
+        describe: "overrides relay chain/runtime",
+      },
+      "port-prefix": {
+        type: "number",
+        default: 34,
+        check: (port) => port >= 0 && port <= 64,
+        describe: "provides port prefix for nodes",
+      },
+    })
+    .help().argv;
+
+  const portPrefix = argv["port-prefix"] || 34;
+  const startingPort = portPrefix * 1000;
+  const parachainName = argv.parachain.toString();
+  const parachain = parachains[parachainName];
+  const parachainChain = argv["parachain-chain"] || parachain.chain;
+
+  const relayName = argv.relay || parachain.relay;
+
+  if (!relayName || !relayNames.includes(relayName)) {
+    console.error(`Invalid relay name: ${relayName}`);
+    console.error(`Expected one of: ${relayNames.join(", ")}`);
+    return;
+  }
+
+  const relay = relays[relayName];
+  const relayChain = argv["relay-chain"] || relay.chain;
+
+  console.log(
+    `🚀 Relay:     ${relayName.padEnd(20)} - ${relay.docker || relay.binary} (${relayChain})`
+  );
+
+  let parachainBinary;
+  if (parachain.binary) {
+    parachainBinary = parachain.binary;
+    const parachainPath = path.join(__dirname, parachain.binary);
+    if (!fs.existsSync(parachainPath)) {
+      console.log(`     Missing ${parachainPath}`);
+      return;
+    }
+  } else {
+    parachainBinary = `build/${parachainName}/moonbeam`;
+    const parachainPath = path.join(__dirname, `build/${parachainName}/moonbeam`);
+    if (!fs.existsSync(parachainPath)) {
+      console.log(`     Missing ${parachainBinary} locally, downloading it...`);
+      child_process.execSync(`mkdir -p ${path.dirname(parachainPath)} && \
+          docker create --name moonbeam-tmp ${parachain.docker} && \
+          docker cp moonbeam-tmp:/moonbeam/moonbeam ${parachainPath} && \
+          docker rm moonbeam-tmp`);
+      console.log(`     ${parachainBinary} downloaded !`);
+    }
+  }
+  console.log(
+    `🚀 Parachain: ${parachainName.padEnd(20)} - ${
+      parachain.docker || parachain.binary
+    } (${parachainChain})`
+  );
+
+  let relayBinary;
+  if (relay.binary) {
+    relayBinary = relay.binary;
+    const relayPath = path.join(__dirname, relay.binary);
+    if (!fs.existsSync(relayPath)) {
+      console.log(`     Missing ${relayPath}`);
+      return;
+    }
+  } else {
+    relayBinary = `build/${relayName}/polkadot`;
+    const relayPath = path.join(__dirname, `build/${relayName}/polkadot`);
+    if (!fs.existsSync(relayPath)) {
+      console.log(`     Missing ${relayBinary} locally, downloading it...`);
+      child_process.execSync(`mkdir -p ${path.dirname(relayPath)} && \
+          docker create --name polkadot-tmp ${relay.docker} && \
+          docker cp polkadot-tmp:/usr/local/bin/polkadot ${relayPath} && \
```

Comment:
The crux of the docker support is copying the compiled binary out of the container and running it locally. Is that going to work on non-linux?

I'm fine if it doesn't, but definitely worth noting.

Reply:
You are right, I'll add information/warning about it